vmhost/contexts/output.go

```diff
@@ -33,8 +34,9 @@ func NewOutputContext(host vmhost.VMHost) (*outputContext, error) {
 	}
 
 	context := &outputContext{
-		host:       host,
-		stateStack: make([]*vmcommon.VMOutput, 0),
+		host:             host,
+		stateStack:       make([]*vmcommon.VMOutput, 0),
+		crtTransferIndex: 0,
```

Comment:
you need to reset the crtTransferIndex on InitState. That is called only once per transaction.

Reply:
good catch